sdk/src/azure/core/az_span_private.h

```diff
@@ -65,7 +65,7 @@ AZ_NODISCARD az_result _az_span_replace(
     int32_t end,
     az_span replacement);
 
-typedef bool (*_az_predicate)(az_span slice);
+typedef bool (*_az_predicate)(uint8_t next_byte);
```

Comment:
This change results in code size wins.

**Before:**
az_span.o 176

**After:**
az_span.o 84

```C
AZ_NODISCARD AZ_INLINE bool _az_is_char(az_span slice, uint8_t c)
{
  return az_span_ptr(slice)[0] == c;
}

static AZ_NODISCARD bool _az_is_new_line(az_span slice) { return _az_is_char(slice, '\n'); }

int B(void)
{
  az_span response_span = AZ_SPAN_FROM_STR( //
        "HTTP/1.2 404 We removed the\tpage!\r\n"
        "\r\n"
        "But there is somebody. :-)");
    az_http_response response = { 0 };
    az_result result = az_http_response_init(&response, response_span);
    
   int32_t offset = 0;
   result = _az_span_scan_until(response_span, _az_is_new_line, &offset);
   printf("%d\n", result);
   
  return 0;
}
```